bin/clonetb

```diff
@@ -0,0 +1,116 @@
+#!/bin/sh
+
+
+# Copyright 2023 Silicon Laboratories Inc.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
+#
+# Licensed under the Solderpad Hardware License v 2.1 (the "License"); you may
+# not use this file except in compliance with the License, or, at your option,
+# the Apache License version 2.0.
+#
+# You may obtain a copy of the License at
+# https://solderpad.org/licenses/SHL-2.1/
+#
+# Unless required by applicable law or agreed to in writing, any work
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Description:
+#   This script is used to clone core-specific verification environments.
+#   (E.g. "core-v-verif/cv32e40x/".)
+#   Try running it, and it will assist you with helpful messages.
+#
+# Examples:
+#   "$ ./bin/clonetb"
+#   "$ ./bin/clonetb -x"
+#   "$ CV_CORE=cv32e40x ./bin/clonetb --unignore"
+
+
+usage() {
+  echo "usage: $0 [-x] [--clone] [--ignore] [--unignore]"
+  echo "  -x          clone cv32e40x subtree"
+  echo "  --clone     clone subtree based on env vars"
+  echo "  --ignore    tell git to ignore the cloned subtree"
+  echo "  --unignore  tell git to not ignore the cloned subtree"
+
+  exit 1
+}
+
+check_env_vars_cvcore() {
+  if [ -z "${CV_CORE}" ]; then
+    echo "error: 'CV_CORE' not defined"
+    exit 1
+  fi
+}
+
+check_env_vars() {
+  check_env_vars_cvcore
+
+  if [ -z "${VERIF_ENV_REPO}" ]; then
+    echo "error: 'VERIF_ENV_REPO' not defined"
+    exit 1
+  fi
+
+  if [ -z "${VERIF_ENV_BRANCH}" ]; then
+    echo "error: 'VERIF_ENV_BRANCH' not defined"
+    exit 1
+  fi
+}
+
+clone() {
+  check_env_vars
+
+  rm -rf ${CV_CORE}
+  git clone  -b ${VERIF_ENV_BRANCH}  ${VERIF_ENV_REPO}  ${CV_CORE}
+}
+
+clone_cv32e40x() {
+  CV_CORE=cv32e40x
```

Comment:
Two comments about this:
1. It would ease usability and maintenance if these were clearly visible as global vars at the top of the file.
2. Would it be possible to set things up so that if the user control these via shell env vars or command-line options?

Reply:
> It would ease usability and maintenance if these were clearly visible as global vars at the top of the file.

It would be the opposite of "progressive disclosure".
For normal usage there is no need to know, and for advanced usage it is trivial to figure it out.

Nevertheless, I've added a comment at the top of the file which explains the env vars.
It can be useful when modifying the script at a later time.

--------

> Would it be possible to set things up so that if the user control these via shell env vars or command-line options?

Having two ways of doing it?

I also played with that idea (and know a way to do it), but ultimately I think it adds unnecessary clutter.
Env vars like `CV_CORE` are already in use in core-v-verif.
It's nice and quick and clean and gets things done.

Reply:
>  Having two ways of doing one thing?

We already have that.  The Makefiles are written such that you can either set vars on the command-line or as shell environment variables:
```
$ make test TEST=hello-world SIMULATOR=xrun
```
...is the same as...
```
$ export SIMULATOR=xrun
$ make test TEST=hello-world
```

Reply:
> We already have that. The Makefiles are written such that you can either set vars on the command-line or as shell environment variables
> [...]
> ...is the same as...
> [...]

Oh, I thought you were asking for command-line options, like `$ myprogram --myarg`.
Those two examples you gave are both using environment variables, not command-line options.
Our makefiles are not written to facilitate that, the syntax of make allows macro definitions to be given as arguments.
By default, posix (and gnu) shell allows you to pass environment variables by prepending them to a command, like `$ MYVAR=somevalue mycommand`, and the "Examples:" section of my comment header shows an example of this.